woocommerce/payment-gateway/class-sv-wc-payment-gateway-payment-form.php

```diff
@@ -991,6 +992,31 @@ public function render_payment_fields() {
 		foreach ( $this->get_payment_fields() as $field ) {
 			$this->render_payment_field( $field );
 		}
+
+		// set the context for the checkout form in case gateways need to reference this in their validation
+		if ($context = $this->getCurrentPaymentFormContext()) {
+			WC()->session->set(
+				sprintf('wc_%s_payment_form_context', $this->get_gateway()->get_id()),
```

Comment:
We can use the extracted `paymentFormContextKeyName` here, ref: https://github.com/gdcorp-partners/woocommerce-gateway-elavon/pull/103#discussion_r1721181978

Reply:
See updates here: https://github.com/godaddy-wordpress/wc-plugin-framework/pull/696/commits/34b27ababe52c14eb16f48d4e90f92628ca6f490

I created a new class for all the logic because I realized that the payment form class isn't accessible in Elavon where we actually want to use this (that's in the `AJAX` class).

Reply:
Good idea, nice